Modules/Loadable/Segmentations/EditorEffects/Python/SegmentEditorThresholdEffect.py

```diff
@@ -109,7 +110,13 @@ def masterVolumeNodeChanged(self):
     masterImageData = self.scriptedEffect.masterVolumeImageData()
     if masterImageData:
       lo, hi = masterImageData.GetScalarRange()
-      self.thresholdSlider.minimum, self.thresholdSlider.maximum = lo, hi
+      if ((lo - self.thresholdSlider.maximum) > math.pow(10, -self.thresholdSlider.decimals)) and ((hi - lo) > math.pow(10, -self.thresholdSlider.decimals)):
+        self.thresholdSlider.maximum = hi
+        self.thresholdSlider.minimum = lo
+      else :
+        self.thresholdSlider.minimum = lo
+        self.thresholdSlider.maximum = hi
```

Comment:
Exactly the same code is executed in the two branches, only in reversed order. Does it change anything?

Reply:
yeah the point is exactly that.  In the first case the minimum is major than the current self.thresholdSlider.maximum and therefore you have to set first the maximum with hi and then you can set the minimum (with the other order will crush).  In the else case is viceversa.

Ideally the best will be to have the setRange method also in Python that handle this automatically...
However, I don't understand why the setRange method is not available in Python.

P.S..: this is a bug in Qt5, I didn't experienced in Qt4.

Reply:
Thanks for the explanation! In that case I think this change is fine. Can you, however, add a comment that nicely explains this? Others looking at the code will have the exact same question as I did, and someone may just change it back silently, saying that "it does the same anyway". Thanks!

Reply:
Only Q_INVOKABLE method, property get/set methods, and slots are python-wrapped. setRange(double min, double max) is none of them. To make it available from Python, you just have to add Q_INVOKABLE.

@Punzo Could you try if it all works well if you call setRange from Python?

Reply:
@lassoan  thanks for the explanation, I didn't know. I checked and if using setRange in python will work as well. I changed the commit and I'll make a pull request to CTK repository as well.